lib/Backend/arm64/LowerMD.cpp

```diff
@@ -1353,89 +1341,95 @@ LowererMD::LowerEntryInstr(IR::EntryInstr * entryInstr)
 
     IR::RegOpnd * r17Opnd = nullptr;
 
-    /*
-    Homed arguments (r0-r7)
-    Callee-Saved float Registers (d16-d29)
-    Callee-Saved Int Registers (r19-r28)
-    Local Variables
-    lr - link Register (r30)
-    fp - Frame Pointer (r29)
-    alloca area
-    */
-
     // Zero-initialize dedicated arguments slot
     if (hasCalls)
     {
         //R17 acts a dummy zero register which we push to arguments slot
         //mov r17, 0
         Assert(r17Opnd == nullptr);
         r17Opnd = IR::RegOpnd::New(nullptr, SCRATCH_REG, TyMachReg, this->m_func);
-        IR::Instr * instrMov = IR::Instr::New(Js::OpCode::MOV, r17Opnd, IR::IntConstOpnd::New(0, TyMachReg, this->m_func), this->m_func);
+        IR::Instr * instrMov = IR::Instr::New(Js::OpCode::MOVZ, r17Opnd, IR::IntConstOpnd::New(0, TyMachReg, this->m_func), this->m_func);
         insertInstr->InsertBefore(instrMov);
         IR::LabelInstr *prologStartLabel = IR::LabelInstr::New(Js::OpCode::Label, this->m_func);
         insertInstr->InsertBefore(prologStartLabel);
         this->m_func->m_unwindInfo.SetPrologStartLabel(prologStartLabel->m_id);
     }
 
+    int32 regPushedSize = 0;
+
     if (!paramRegs.IsEmpty())
     {
         for (int i = LAST_INT_ARG_REG; i >= FIRST_INT_ARG_REG; --i)
```

Comment:
I think you need a TODO: here to indicate that we will eventually push in pairs. (Unless I missed that code.)

Reply:
That is the plan. Added a ToDo

---
In reply to: [143538276](https://github.com/Microsoft/ChakraCore/pull/3903#discussion_r143538276) [](ancestors = 143538276)